File: PythonP/06_상속/Ex06_05_super.py
'''
Created on 2022. 5. 26.

@author: admin
'''

class Parent:
    x = 10    
    def method(self):
        print(self)
        print(f"P_x : {self.x}")
        
class Child(Parent):
    x = 20    
    def method(self):
        x = 30
        print(f"x : {x}")
        print(f"self.x : {self.x}")
        print(f"super().x : {super().x}")
        super().method()
        
c = Child()
c.method()

File: 06_상속/src/Ex06_05_super.java

class Parent {
	int x = 0;
	void method() {
		System.out.println("P_x : " + x);
	}
} // Parent

class Child extends Parent {
	int x = 20;
	void method() {
		int x = 30;	// 지역변수
		System.out.println("x : " + x);
		System.out.println("this.x : " + this.x);
		System.out.println("super.x : " + super.x);
		
		super.method();
	}
} // Child

public class Ex06_05_super {
	public static void main(String[] args) {

		Child c = new Child();
		c.method();
		
	}
}
